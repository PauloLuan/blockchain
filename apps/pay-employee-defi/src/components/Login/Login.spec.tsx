import React from 'react'
import { render } from '@testing-library/react'

import { Login } from '.'

describe('Login Page', () => {
  it('should renders login page correctly', () => {
    const { asFragment } = render(<Login />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("should find the login test id and it's mindingo's class", () => {
    const { getByTestId } = render(<Login />)
    const result = getByTestId('login').textContent
    expect(result).toEqual('mindingo')
  })
})
