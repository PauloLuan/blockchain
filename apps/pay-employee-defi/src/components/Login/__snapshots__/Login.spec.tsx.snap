// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should renders login page correctly 1`] = `
<DocumentFragment>
  <div
    class="css-h74a88"
    v="100vh"
  >
    <form
      class="css-kwzi0y"
    >
      <div
        class="chakra-stack css-n21gh5"
      >
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-uvho5g"
            for="field-1"
            id="field-1-label"
          >
            Seu Email
          </label>
          <input
            aria-describedby="field-1-helptext"
            class="chakra-input css-0"
            data-testid="login-email-field"
            id="field-1"
            name="name"
            type="email"
          />
          <div
            class="chakra-form__helper-text css-0"
            id="field-1-helptext"
          >
            Pare de usar e-mail da bol
          </div>
        </div>
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-uvho5g"
            for="field-2"
            id="field-2-label"
          >
            Sua senha:
          </label>
          <input
            aria-describedby="field-2-helptext"
            class="chakra-input css-0"
            data-testid="login-password-field"
            id="field-2"
            name="password"
            type="password"
          />
          <div
            class="chakra-form__helper-text css-0"
            id="field-2-helptext"
          >
            Respeitamos a LGPD
          </div>
        </div>
        <button
          class="chakra-button css-6k6edw"
          data-testid="login-submit-button"
          type="button"
        >
          Sign In
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
